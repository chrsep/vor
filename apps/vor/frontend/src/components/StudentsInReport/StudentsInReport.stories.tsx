import { Meta, Story } from "@storybook/react"
import StudentsInReport, { PageManageReportsProps } from "./StudentsInReport"

export default {
  title: "Core/PageManageReports",
  component: StudentsInReport,
  parameters: {
    componentSubtitle: "Just a simple PageManageReports",
  },
} as Meta

const Template: Story<PageManageReportsProps> = (args) => (
  <StudentsInReport {...args} />
)

export const Default = Template.bind({})
Default.args = {}
