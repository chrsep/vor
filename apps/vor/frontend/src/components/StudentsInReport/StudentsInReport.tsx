import { FC } from "react"
import { Box, Text, ThemeUIStyleObject } from "theme-ui"
import { borderBottom } from "../../border"
import { useGetAllStudents } from "../../hooks/api/students/useGetAllStudents"
import { STUDENT_REPORT_URL } from "../../routes"
import { Link } from "../Link/Link"

export interface PageManageReportsProps {
  reportId: string
  studentId: string
  containerSx?: ThemeUIStyleObject
}

const StudentsInReport: FC<PageManageReportsProps> = ({
  reportId,
  studentId,
}) => {
  const students = useGetAllStudents("", true)

  return (
    <Box>
      {students.data?.map(({ id, name }) => (
        <Student
          key={id}
          reportId={reportId}
          studentId={id}
          studentId1={studentId}
          name={name}
        />
      ))}
    </Box>
  )
}

const Student: FC<{
  reportId: string
  studentId: string
  studentId1: string
  name: string
}> = ({ name, reportId, studentId, studentId1 }) => (
  <Link
    to={STUDENT_REPORT_URL(reportId, studentId)}
    sx={{
      width: "100%",
      backgroundColor: studentId1 === studentId ? "primaryLighter" : "",
      display: "flex",
      alignItems: "center",
      ...borderBottom,
      "&:hover": {
        backgroundColor: "primaryLightest",
      },
    }}
  >
    <Text p={3} className="truncate" sx={{ fontSize: 1 }}>
      {name}
    </Text>

    <Box>
      <Text sx={{ fontSize: 1 }}>Joyful 1</Text>
    </Box>
  </Link>
)

export default StudentsInReport
