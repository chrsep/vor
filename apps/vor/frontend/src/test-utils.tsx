/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { setupI18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import Theme from "./gatsby-plugin-theme-ui"
import enCatalog from "../i18n/lingui/en/messages.js"

const i18n = setupI18n()
// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
