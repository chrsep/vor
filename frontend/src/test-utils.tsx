/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <Box fontSize={[16, 20]}>{children}</Box>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
