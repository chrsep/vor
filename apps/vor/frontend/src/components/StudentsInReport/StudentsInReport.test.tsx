import { render } from "../../test-utils"
import StudentsInReport from "./StudentsInReport"

describe("PageManageReports", () => {
  it("should render correctly", () => {
    const { container } = render(<StudentsInReport reportId="" studentId="" />)
    expect(container).toMatchSnapshot()
  })
})
