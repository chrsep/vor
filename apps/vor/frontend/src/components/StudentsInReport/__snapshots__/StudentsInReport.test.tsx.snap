// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageManageReports should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-right-width: 1px;
  border-right-style: solid;
  border-right-color: var(--theme-ui-colors-border);
  width: 100%;
  height: 100vh;
  -webkit-position: sticky;
  position: sticky;
  top: 0;
  overflow-y: auto;
  padding-bottom: 48px;
}

@media screen and (min-width: 72em) {
  .emotion-1 {
    max-width: 340px;
  }
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    />
  </div>
</div>
`;
